tests/linalg_test.py

```diff
@@ -61,13 +61,16 @@ class NumpyLinalgTest(jtu.JaxTestCase):
       for rng in [jtu.rand_default()]))
   def testCholesky(self, shape, dtype, rng):
     def args_maker():
-      a = rng(shape, dtype)
+      factor_shape = shape[:-1] + (2 * shape[-1],)
```

Comment:
This improves the matrix conditioning somehow? I'm trying to follow the math here.

Reply:
Yeah, this random variable is Wishart distributed and this change increases the degrees of freedom. One way to think about it is that if the trailing two dimensions are (n, n), then the matrix is just barely full-rank, since (n, n-1) would be rank-deficient. Another way to think about it if the trailing two dimensions are (n, k) is that this is like a maximum likelihood estimator for a covariance matrix (ie sample covariance matrix) scaled up by a factor of k, and so as we increase k the spectrum will concentrate around the spectrum of k * I where I is the (n, n) identity matrix; thus the condition number should tend to 1 as we increase k.

Reply:
Yet another way to think about it is on the margin: if the trailing dimensions are (n, k) and we consider bumping that to (n, k+1), we're applying X \mapsto X + vv' to the PSD cone, for v ~ N(0, I). That's almost surely a strictly positive map on the PSD cone, and meaning it's a contraction in the Hilbert metric and trajectories will concentrate along the ray { \lambda I : \lambda > 0 }. Here's a nice paper on those notions of contraction: https://arxiv.org/pdf/1503.09113.pdf.